### PR TITLE
Add card layout page to Patterns section

### LINKED ISSUE #10697: [Markdown] transfer Card layout pages
[Doc](https://docs.google.com/document/d/13LrzMI5SMzDZ3Wm117Yz4olE7PzBCFQh2_B5C07WQlE/edit#heading=h.gfe7lmbx5ooo)

### PR DESCRIPTION
**Pull request by sam-b-rose**
Closes #10697 

🎩

- Run locally with `yarn workspace polaris.shopify.com dev` 
- Visit http://localhost:3000/patterns/card-layout

### CODE REVIEW

File: polaris.shopify.com/content/patterns/card-layout/index.mdx [hunk 1]
```diff
@@ -0,0 +1,639 @@
...
+    Use space-400 inside card sections, as it can disconnect content that belongs together. Instead, default to space-200 and upsize to space-300 if the former seems too tight.
+
+  </Medium>
+</LayoutSection>
+
+<LayoutSection card>
+  <Text>
+    ## Alignment
+
+    Content alignment makes cards more balanced and easier for merchants to scan. Content should be vertically aligned along both the left and right edge of the card. It’s always more important to align content [optically](#optically), so that it appears to be aligned rather than technically being aligned.
```
Comment (sam-b-rose) [inline polaris.shopify.com/content/patterns/card-layout/index.mdx:617]: @johanstromqvist this was trying to link to a section in the Google doc, but I wasn't sure which section it needs to be linked

**Review by sam-b-rose — COMMENTED** [review]
@johanstromqvist  I've only added all the structure and spacing content into a single page for now. Seems like the pattern section routing works differently than the other design guidelines content.

I'll look into if we can adjust things to accommodate the tabs, though with the TOC outline on the side the page length doesn't feel too long.

**Comment by johanstromqvist** [comment]
Thanks tons, Sam!

There are a few outstanding issues. Do you want me to create separate issues for those?

<details><summary>Routing</summary>
>  I've only added all the structure and spacing content into a single page for now. Seems like the pattern section routing works differently than the other design guidelines content.

Good call! The routing problem goes beyond these guidelines. There's lots of patterns incoming and it's important that the experience, contribution and maintenance is high quality and the same as the Design section. So this seems important to fix sooner rather than later. What do you think?

</details> 
<details><summary>Images appear cropped</summary>
image
</details> 
<details><summary>Missing headlines</summary>
A consequence of the single page approach is that the main headlines `Structure` and `Spacing` are missing. This makes the hierarchy messy and the content loses some important context. Not sure what a reasonable short term solution is (assuming the long term solution is tabs). Could they perhaps be added as `H2s` without card containers, while current `H2s` drop to `H3s`, etc?
</details> 
<details><summary>Missing link to "optical"</summary>
The section appears to be gone. I've asked the in Slack about its whereabouts. Can be ignored for now.
https://web.archive.org/web/20230913023754/https://polaris.shopify.com/design/space#optical-adjustment
</details>

**Pull request merged**